From 4f1e0b355a568cbfd406ec95a375e44d42904cdc Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 22 Mar 2019 16:52:27 +0100
Subject: [PATCH] lib: Always try to pair pointer devices

Back in the heady Bluetooth 2.0 days, the majority of pointer devices
could work without pairing, and some did not even support pairing.

But pointer devices really want pairing for all the subsequent versions
(from "Secure Simple Pairing" in Bluetooth 2.1, 2007), so always try to
pair pointer devices.
---
 lib/pin-code-database.xml | 26 --------------------------
 1 file changed, 26 deletions(-)

Index: gnome-bluetooth-3.28.0/lib/pin-code-database.xml
===================================================================
--- gnome-bluetooth-3.28.0.orig/lib/pin-code-database.xml
+++ gnome-bluetooth-3.28.0/lib/pin-code-database.xml
@@ -61,22 +61,6 @@
 	     with a OUI from Alps Electric -->
 	<device vendor="alps" name="BD Remote Control" pin="NULL"/>
 
-	<!-- Apple Wireless and Mighty Mouse
-	     Note: Apple doesn't follow the specs, and requires their mice
-	     to be paired -->
-	<device vendor="apple" type="mouse" pin="0000"/>
-
-	<!-- Apple Magic Trackpad
-	     Note: same as above, tablet needs to be paired -->
-	<device vendor="apple" type="tablet" pin="0000"/>
-
-	<!-- Microsoft mice
-	     Again, same as above -->
-	<device oui="7C:1E:52:" type="mouse" name="Microsoft Wedge Touch Mouse" pin="0000"/>
-	<device oui="00:1D:D8:" type="mouse" name="Microsoft Wireless Notebook Presenter Mouse 8000" pin="0000"/>
-	<device oui="28:18:78:" type="mouse" name="Microsoft Sculpt Comfort Mouse" pin="0000"/>
-	<device oui="30:59:B7:" type="mouse" name="Microsoft Sculpt Comfort Mouse" pin="0000"/>
-
 	<!-- BT-GPS8 JENTRO, GPS mouse -->
 	<device oui="00:0D:B5:" name="BT-GPS8 JENTRO" pin="0000"/>
 
@@ -86,9 +70,6 @@
 	<!-- ION iCade Game Controller -->
 	<device name="iCade" type="keyboard" pin="ICADE"/>
 
-	<!-- Logitech Ultrathin Touch Mouse -->
-	<device oui="00:1F:20:" name="Ultrathin Touch Mouse" pin="0000"/>
-
 <!-- GPS devices -->
 	<device oui="00:0D:B5:" name="TomTom Wireless GPS MkII" pin="0000"/>
 	<device oui="00:0D:B5:" name="GPS-GW-005" pin="0000"/>
@@ -161,9 +142,6 @@
 
 <!-- Generic types -->
 
-	<!-- Mice don't usually need pincodes -->
-	<device type="mouse" pin="NULL"/>
-
 	<!-- Printers -->
 	<device type="printer" pin="0000"/>
 
